src/libcore/iter/traits/iterator.rs

```diff
@@ -1078,15 +1075,14 @@ pub trait Iterator {
     /// #![feature(iter_map_while)]
     /// use std::convert::TryFrom;
     ///
-    /// let a = [0, -1, 1, -2];
+    /// let a = [0, 1, 2, -3, 4, 5, -6];
     ///
-    /// let mut iter = a.iter().map_while(|x| u32::try_from(*x).ok());
+    /// let iter = a.iter().map_while(|x| u32::try_from(*x).ok());
+    /// let vec = iter.collect::<Vec<_>>();
     ///
-    /// assert_eq!(iter.next(), Some(0u32));
-    ///
-    /// // We have more elements that are fit in u32, but since we already
-    /// // got a None, map_while() isn't used any more
-    /// assert_eq!(iter.next(), None);
+    /// // We have more elements which could fit in u32 (4, 5), but `map_while`
+    /// // has stopped on the first `None` from predicate (when working with `-3`)
```

Comment:
Being pedantic here again, `map_while` didn't stop, but `collect` did. So maybe something like this?

> ..., but `map_while` returned `None` for `-3` (as the predicate returned `None`) and `collect` stops at the first `None` entcountered.

Reply:
Seems reasonable. At the time of writing this comment, I was thinking that `map_while` _is_ fused :sweat_smile: